### ISSUE TITLE
Infovore compiles but doesn't run successfully on Ubuntu 12.04.1 LTS

### ISSUE THREAD

**Issue by jeffreycrow** [paulhoule/infovore#1]
I'm trying to run infovore on Ubuntu 12.04.1 LTS. For Java, I'm using openjdk-7-jdk. I have tried v1.1, v1.1.1 and a current git snapshop. The first two compiled, the git snapshot didn't, and none of them would run. I've tried following the instructions, but I'm a web developer and not knowledgeable in Java, so I am at a bit of a loss. Below is my terminal commands and the errors outputted.

~/infovore$ mvn clean install
~/infovore$ source infovore-apps/path.sh
~/infovore$ infovore run partitionFreebaseRDF ../freebase-rdf-2013-04-07-00-00.gz baseKBLime
Exception in thread "main" java.lang.UnsupportedClassVersionError: com/ontology2/hydroxide/shell/InfovoreShell : Unsupported major.minor version 51.0
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:634)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
    at java.net.URLClassLoader.defineClass(URLClassLoader.java:277)
    at java.net.URLClassLoader.access$000(URLClassLoader.java:73)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:212)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:205)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:321)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:294)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:266)
Could not find the main class: com.ontology2.hydroxide.shell.InfovoreShell. Program will exit.

Would love any help on getting infovore up and running as I'm quite interested in using freebase for a project I'm working on.

**Comment by paulhoule** [comment]
I'll investigate this.

**Comment by jeffreycrow** [comment]
I got this working on an Ubuntu Server EC2 instance so whatever was happening has to be some quirk of my local install.

**Comment by gdiazc** [comment]
I have the exact same problem on a Macbook Pro running OS X 10.7. Any ideas?

**Comment by paulhoule** [comment]
Are you sure that you're using JDK 7 to run it,  not just to build it?

**Comment by gdiazc** [comment]
Hi, you are right. This problem is caused when you compile the code with one version of Java and execute with a different version. The problem was the following:

"javac -version" gives "javac 1.7.0_21", which is good.
"java -version" gives "java version 1.7.0_21", which is also good.

These are installed in "usr/bin/java"

The problem is that the infovore script (infovore-apps/target/appassembler/bin/infovore) uses the System java, which is an older version.

I found this by editing the infovore script and inserting the line "echo $JAVACMD" before the "exec" line, and I got:

"/System/Library/Frameworks/JavaVM.framework/Versions/CurrentJDK/Home/bin/java"

This executable is actually version 1.6...

So, I added the line "JAVACMD='java'" (java is surrounded by single quotes) before the exec line and it worked well.

I hope this is useful to anyone who had the same problem!
